proposals/bootstrapper-proposal.md

```diff
@@ -0,0 +1,253 @@
+# Kubeflow Bootstrapping Modifications
+
+## Motivation
+
+### Current Kubeflow bootstrapper requires cluster-admin privileges
+The current bootstrapper generates a kubeflow application into a PersistentVolume and will optionally deploy it to the API-server. The bootstrapper uses ksonnet to generate the kubernetes manifests that define the kubeflow runtime. The manifest generation by ksonnet is ordered across phases:
+- optional creation of a Namespace
+- creation of ClusterRoles and ClusterRoleBindings
+- creation of namespace scoped resources (services, serviceaccounts, ...)
+- creation of IAP/GCP single signon
+
+The bootstrapper is run as cluster-admin although not all phases above require cluster-admin privileges.
+
+### RBAC rules for data scientists aren't utilized
+A data scientist is authenticated either as an admin or a google user (via IAP) but service account RBACs are used when running processes. Actions and their RBACs are shown below:
+
+| Action | ClusterRole | ClusterRoleBinding | ServiceAccount |
+| :---: | :---: | :---: | :---: |
+| train | tf-job-operator | tf-job-operator | tf-job-operator |
+| job-dashboard | tj-job-dashboard | tf-job-dashboard | tf-job-dashboard |
+| central-ui | centraldashboard | centraldashboard | centraldashboard |
+| nfs | system:persistent-volume-provisioner | nfsnfs-provisioner | kubeflow:nfs |
+| IAP | cloud-endpoints-controller | cloud-endpoints-controller | kubeflow:cloud-endpoints-controller |
+
+
+## Goals
+Divide bootstrapper into 2 phases: an initialization phase and a deployment phase. Each phase will run with different authorizations/users.
```

Comment:
yes i think it was a good choice to have bootstrapper optionally deploy so that other deployment tools could possibly be used. The gist of the proposal is to allow data scientists to deploy w/o cluster-admin or cluster wide privileges and to have data-scientists use their own service accounts to allocate pods.